### DISCUSSION TITLE
Bandwidth and storage

### DISCUSSION THREAD

**Discussion by PeterW-DASA** [m-ld/m-ld-spec#46] [Discussion]
Is it the case that all nodes eventually end up holding a full copy of all the data, and if so, what are the storage and bandwidth implications?

**Comment by gsvarovsky** [comment]
Hi Peter, thanks for the question.

It's stronger than that, actually – a new clone starts by streaming in the data from its peers, so every clone has a full copy. However, the implications of this critically depend on what is meant by 'all of the data'. With m-ld it's important to consider what the unit of synchronisation is, in your overall data model. Taking Google Docs as an example, you have each document as one unit. In a large-scale system, you would not have one m-ld domain for the whole system. Instead, you'd partition the data into synchronised units.

We do want to extend the model in future to allow for 'sub-domain' clones, which only replicate some part of the domain and so avoid any low storage limits on some devices. Watch out for a blog article on this very subject, coming soon 😀
